bugsnag-android-core/src/androidTest/java/com/bugsnag/android/ClientConfigTest.java

```diff
@@ -38,7 +39,7 @@ public void tearDown() {
     public void testSetContext() throws Exception {
         client.setContext("JunitTest");
         assertEquals("JunitTest", client.getContext());
-        assertEquals("JunitTest", config.getContext());
+        assertNull(config.getContext());
```

Comment:
I don't understand why this test how this test even passed before! Isn't the purpose of this test script to ensure that values passed to config are passed to client? Therefore shouldn't we be setting it on config and asserting on client (like the delivery test below)?

Reply:
The test passed previously because the value of `context` on `Client was backed by a field on `Configuration`. That's not the case anymore as we take a deep-copy of `Configuration`, so the field can be asserted as null here.

Reply:
OK, but would a more useful test not be to verify the other way round? However this is not just for context, but for the whole client:config relationship, so I think we could do with an overall review of unit test structure at the end of this refactor.